internal/domain/transitions.go

```diff
@@ -554,53 +549,66 @@ func doTransition(transitionID uuid.UUID) {
 		// Check to see if we got everything back. ReserveComponents returns
 		// everything that either was successfully reserved or had a valid
 		// deputy key.
-		if len(resData) != len(reservationData) {
-			resMap := make(map[string]*hsm.ReservationData)
-			for _, res := range reservationData {
-				resMap[res.XName] = res
-			}
-			for _, res := range resData {
-				if _, ok := resMap[res.XName]; !ok {
-					var depErrMsg string
-					var powerAction string
-					comp := xnameMap[res.XName]
-					comp.Task.Status = model.TransitionTaskStatusFailed
-					if res.DeputyKey == "" {
-						// TODO: Check ExpirationTime and wait to try again?
-						//       Could be that we restarted and just need to
-						//       wait for the old locks to fall off.
-						comp.Task.Error = "Unable to reserve component"
-						depErrMsg = fmt.Sprintf("Unable to reserve dependent component, %s.", comp.Task.Xname)
-					} else {
-						// We were given a deputy key that was invalid so we tried
-						// reserving the component but we failed to reserve it.
-						comp.Task.Error = "Invalid deputy key and unable to reserve component"
-						depErrMsg = fmt.Sprintf("Invalid deputy key and unable to reserve dependent component, %s.", comp.Task.Xname)
-					}
-					comp.Task.StatusDesc = "Failed to achieve transition"
-					switch(comp.Task.Operation) {
-					case model.Operation_SoftRestart: fallthrough
-					case model.Operation_HardRestart: fallthrough
-					case model.Operation_SoftOff: fallthrough
-					case model.Operation_Off:
+		resMap := make(map[string]*hsm.ReservationData)
```

Comment:
The vars resMap, reservationData, and resData could be named different.

resMap could be changed to reservationMap given that it contains reservationData and not resData. And use reservation when it comes from reservationData and res when from resData.

Alternately reservationData and resMap could be named for what type of reservation they are (deputyReservation? reservedReservation? successfulReservation?)

But only make a change if you agree, and variable names are important to you. Even I wouldn't do what I think if I could think something else.